runelite-client/src/main/java/net/runelite/client/plugins/cleanermenus/CleanerMenusPlugin.java

```diff
@@ -0,0 +1,57 @@
+package net.runelite.client.plugins.cleanermenus;
+
+import com.google.common.eventbus.Subscribe;
+import net.runelite.api.Client;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.events.MenuOpened;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+import javax.inject.Inject;
+import java.util.AbstractMap;
+import java.util.HashMap;
+
+@PluginDescriptor(name = "Cleaner Menus")
+public class CleanerMenusPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Subscribe
+	public void onMenuOpened(MenuOpened event)
+	{
+		//      target          option                                                one to show| position| number of times shown
+		HashMap<String, HashMap<String, AbstractMap.SimpleEntry<AbstractMap.SimpleEntry<MenuEntry, Integer>, Integer>>> repeated = new HashMap<>();
+
+		int count = 0;
+		for (MenuEntry menuEntry : event.getMenuEntries())
+		{
+			if (!repeated.containsKey(menuEntry.getTarget()))
+			{
+				repeated.put(menuEntry.getTarget(), new HashMap<>());
+			}
+
+			if (!repeated.get(menuEntry.getTarget()).containsKey(menuEntry.getOption()))
+			{
+				repeated.get(menuEntry.getTarget()).put(menuEntry.getOption(), new AbstractMap.SimpleEntry<>(new AbstractMap.SimpleEntry<>(menuEntry, count), 0));
+				count++;
+			}
+
+
+			repeated.get(menuEntry.getTarget()).get(menuEntry.getOption()).setValue(repeated.get(menuEntry.getTarget()).get(menuEntry.getOption()).getValue() + 1);
+		}
+
+		MenuEntry[] toShow = new MenuEntry[count];
+
+		repeated.forEach((k1, v1) -> v1.forEach((k2, v2) ->
```

Comment:
The k1, v1, k2, v2 variables should be renamed to to what they are supposed to represent, similar to the comment.